CMakeLists.txt

```diff
@@ -23,103 +23,128 @@ string(REGEX REPLACE ${VERSION_REGEX} "\\1" VERSION_STRING "${VERSION_STRING}")
-
-        if(CLANG_TIDY_EXE)
-            if(CLANG_TIDY_FIX)
-                set(DO_CLANG_TIDY "${CLANG_TIDY_EXE}" "-fix")
-            else()
-                set(DO_CLANG_TIDY "${CLANG_TIDY_EXE}")
-            endif()
-        endif()
-    endif()
-
-    if(NOT CMAKE_VERSION VERSION_LESS 3.9)
-        find_package(Doxygen)
-        if(Doxygen_FOUND)
-            add_subdirectory(docs)
-        else()
-            message(STATUS "Doxygen not found, not building docs")
-        endif()
-    else()
-        message(STATUS "Newer CMake adds Doxygen support, update CMake for docs")
-    endif()
+find_program(CLI11_CLANG_TIDY_EXE NAMES clang-tidy DOC "Path to clang-tidy")
+find_package(Doxygen)
+
+list(APPEND force-libcxx "CMAKE_CXX_COMPILER_ID STREQUAL \"Clang\"")
+list(APPEND force-libcxx "CMAKE_SYSTEM_NAME STREQUAL \"Linux\"")
+list(APPEND force-libcxx "CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME")
+
+list(APPEND clang-tidy-fix "CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME")
+list(APPEND clang-tidy-fix "CMAKE_VERSION VERSION_GREATER 3.5")
+list(APPEND clang-tidy-fix "CLI11_CLANG_TIDY_EXE")
+list(APPEND clang-tidy-fix "CLI11_BUILD_EXAMPLES")
+
+list(APPEND build-docs "CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME")
+list(APPEND build-docs "CMAKE_VERSION VERSION_GREATER 3.10")
+list(APPEND build-docs "Doxygen_FOUND")
+
+option(CLI11_WARNINGS_AS_ERRORS "Turn all warnings into errors (for CI)")
+option(CLI11_SINGLE_FILE "Generate a single header file")
+
+cmake_dependent_option(CLI11_BUILD_DOCS
+  "Build CLI11 documentation" ON
+  "${build-docs}" OFF)
+
+cmake_dependent_option(CLI11_BUILD_TESTS
+  "Build CLI11 tests" ON
+  "BUILD_TESTING;CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME" OFF)
+
+cmake_dependent_option(CLI11_BUILD_EXAMPLES
+  "Build CLI11 examples" ON
+  "CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME" OFF)
+
+cmake_dependent_option(CLI11_BUILD_EXAMPLE_JSON
+  "Build CLI11 json example" OFF
+  "CLI11_BUILD_EXAMPLES" OFF)
+
+cmake_dependent_option(CLI11_SINGLE_FILE_TESTS
+  "Duplicate all the tests for a single file build" OFF
+  "BUILD_TESTING;CLI11_SINGLE_FILE" OFF)
+
+cmake_dependent_option(CLI11_INSTALL
+  "Install the CLI11 folder to include during install process" ON
+  "CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME" OFF)
+
+cmake_dependent_option(CLI11_CLANG_TIDY_FIX
+  "Perform fixes for clang-tidy" OFF
+  "${clang-tidy-fix}" OFF)
+
+cmake_dependent_option(CLI11_FORCE_LIBCXX
+  "Force clang to use libc++ instead of libstdc++ (Linux only)" OFF
+  "${force-libcxx}" OFF)
+
+if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND NOT DEFINED CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 11)
+endif()
+
+if (NOT DEFINED CMAKE_CXX_EXTENSIONS)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+
+if (NOT DEFINED CMAKE_CXX_STANDARD_REQUIRED)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+
+if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 else()
-    set(CUR_PROJ OFF)
+  get_property(cli11-use-folders GLOBAL PROPERTY USE_FOLDERS)
```

Comment:
Why are you providing a way to turn this off? Why would you turn it off?

Reply:
Oh this might have been a small check on my part and I forgot to remove the whole `else()` block 😅